File: VisaD.Application/Applications/Dtos/LanguageProficiencyDto.cs
using VisaD.Application.Nomenclatures.Dtos;
using VisaD.Data.Nomenclatures;

namespace VisaD.Application.Applications.Dtos
{
	public class LanguageProficiencyDto
	{
		public int Id { get; set; }
		public NomenclatureDto<Language> Language { get; set; }
		public NomenclatureDto<LanguageDegree> Reading { get; set; }
		public NomenclatureDto<LanguageDegree> Writing { get; set; }
		public NomenclatureDto<LanguageDegree> Speaking { get; set; }
	}
}

File: WebApplication/src/modules/application/models/language-proficiency.dto.ts
import { NomenclatureDto } from 'src/modules/nomenclature/common/models/nomenclature-dto';

export class LanguageProficiencyDto {
  id: number;
  language: NomenclatureDto;
  reading: NomenclatureDto;
  writing: NomenclatureDto;
  speaking: NomenclatureDto;
  disable: boolean = false;
  required: boolean = false;
}
